### PR TITLE
Euen.2.rebar3

### LINKED ISSUE #2: Move from erlang.mk to rebar3
Migrate to rebar3 as described [here](https://github.com/inaka/guidelines/blob/master/ERLANG.md#project-setup).

This is requested by esl/TODO#62

### PR DESCRIPTION
**Pull request by cabol**
[Close #2] Replace erlang.mk by rebar3
 - Fix sumo_store_riak to support terms as custom data type.
 - Fix tests according with new sumo changes.
 - Remove erlang.mk related files.
 - Fix rebar.config

### CODE REVIEW

File: src/sumo_store_riak.erl [hunk 1]
```diff
@@ -417,39 +424,57 @@ validate_date({FieldType, _, FieldValue}) ->
...
+  sleep_custom(FieldValue, Type);
+sleep_fun(_, _, FieldValue, _) ->
   FieldValue.
 
 %% @private
-wakeup(Doc) ->
-  sumo_utils:doc_transform(fun wakeup_fun/1, Doc).
+sleep_custom(FieldValue, term) ->
+  base64:encode(term_to_binary(FieldValue));
+sleep_custom(FieldValue, _) ->
```
Comment (elbrujohalcon) [inline src/sumo_store_riak.erl]: Doesn't this defeat the nested field search capabilities of this backend? Shouldn't we implement something a little bit smarter to store nested docs?

File: test/test.config [hunk 2]
```diff
@@ -23,10 +23,9 @@
        {get_options, [{r, 2}, {pr, 0}]}
      ]}
    ]},
-
    {docs, [
-     {sumo_test_people_riak, sumo_test_riak},
-     {sumo_test_purchase_order, sumo_test_riak}
+     {people, sumo_test_riak, #{module => sumo_test_people_riak}},
+     {purchase, sumo_test_riak, #{module => sumo_test_purchase_order}}
```
Comment (elbrujohalcon) [inline test/test.config]: Should be `purchases`, plural.

Comment (cabol) [inline test/test.config]: I think this is totally subjective, what I've seen is that it is named in the same way as its entity, in this case `purchase_order`, but the idea is to allow the users to put what they consider (`purchases`, `purchase`, etc.). We can change it, but again, IMHO it is subjective.

Comment (elbrujohalcon) [inline test/test.config]: It's, but in this case `people` is plural already and consistency is something we value. Do you want to switch that one to `person` there?
FWIW, in my mind `sumo:find_all(people).` looks much better than `sumo:find_all(person).` ¯\\\_(ツ)_/¯

Comment (cabol) [inline test/test.config]: Ok I see, I got your point, in that case you're right, I agree it should be named `purchases`

**Pull request merged**